src/main/java/com/example/demo/repository/TeamRepository.java

```diff
@@ -0,0 +1,59 @@
+package com.example.demo.repository;
+
+import com.example.demo.model.Team;
+import com.example.demo.tournament.CSVcontroller;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class TeamRepository implements BaseTeamRepository {
+
+    private static final List<Team> teams;
+
+    static {
+        teams = new ArrayList<>();
+        CSVcontroller.Read(teams);
```

Comment:
this CSVController should be a spring component and injected in this class instead of using static. static usage creates high coupling and doesn't benefit from what Spring framework offers